chain/bitcoind_client.go

```diff
@@ -1450,3 +1413,58 @@ func (c *BitcoindClient) LookupInputMempoolSpend(op wire.OutPoint) (
 
 	return c.chainConn.events.LookupInputSpend(op)
 }
+
+// resetWatchedFilters empties the maps used to track outpoints, addresses, and
+// txns.
+func (c *BitcoindClient) resetWatchedFilters() {
+	c.watchMtx.Lock()
+	defer c.watchMtx.Unlock()
+
+	c.watchedOutPoints = make(map[wire.OutPoint]struct{})
+	c.watchedAddresses = make(map[string]struct{})
+	c.watchedTxs = make(map[chainhash.Hash]struct{})
+}
+
+// updateWatchedFilters is used to update the internal maps that track the
+// watched addresses, outpoints, or txns.
+func (c *BitcoindClient) updateWatchedFilters(update any) {
+	c.watchMtx.Lock()
+	defer c.watchMtx.Unlock()
+
+	switch update := update.(type) {
+	// We're adding the addresses to our filter.
+	case []btcutil.Address:
+		for _, addr := range update {
+			c.watchedAddresses[addr.String()] = struct{}{}
+		}
+
+	// We're adding the outpoints to our filter.
+	case []wire.OutPoint:
+		for _, op := range update {
+			c.watchedOutPoints[op] = struct{}{}
+		}
+
+	case []*wire.OutPoint:
```

Comment:
seems like a design flaw to have `*wire.OutPoint` and `wire.OutPoint`

Reply:
yeah, the code is pretty old, which will be fixed in an upcoming refactor.